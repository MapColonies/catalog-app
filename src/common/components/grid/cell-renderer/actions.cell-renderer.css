#actionsCellRenderer.actionsContainer {
  justify-content: flex-end;
  position: relative;
  display: flex;
  top: 0;
  right: 0;
  width: 100px;
  /*background-color: rgb(44,65,89);*/
  border-radius: 50px;
  /* 
  ColorRes = Color * alpha + Background * (1 - alpha); 
    Color = rgba(33, 150, 243, 0.1)
    Background = rgb(45, 55, 72)

    ColorRes.Red   =   33 * 0.1 + 45 * 0.9
    ColorRes.Green =  150 * 0.1 + 55 * 0.9
    ColorRes.Blue  =  243 * 0.1 + 72 * 0.9
  */
}

body[dir='rtl'] #actionsCellRenderer.actionsContainer {
  right: unset;
  left: 0;
}

#actionsCellRenderer.actionsContainer .actionIcon {
  right: unset;
  left: unset;
}

body[dir='rtl'] #actionsCellRenderer.actionsContainer .actionIcon {
  left: unset;
  right: unset;
}

#gridActionsMenuContainer .mdc-menu-surface.mdc-menu-surface--open {
  top: 42px !important;
  left: -128px !important;
  width: 240px;
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
  position: fixed;
}

body[dir='rtl'] #gridActionsMenuContainer .mdc-menu-surface.mdc-menu-surface--open {
  right: -128px !important;
  left: unset;
}

#gridActionsMenuContainer {
}

body[dir='rtl'] #gridActionsMenuContainer {
  right: 20px;
}

#actionsCellRenderer .actionMenuItem {
  display: flex;
  align-items: center;
  cursor: pointer;
}

#actionsCellRenderer .mdc-list-item {
  padding: 0 4px;
}

#actionsCellRenderer #allActionsIcon {
  position: relative;
  top: 6px;
}

#actionsCellRenderer .actionMenuItemTitle {
  font-size: 16px;
  padding-left: 8px;
}

body[dir='rtl'] #actionsCellRenderer .actionMenuItemTitle {
  padding-left: unset;
  padding-right: 8px;
}