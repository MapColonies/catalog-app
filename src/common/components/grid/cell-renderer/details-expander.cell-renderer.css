.expanderContainer{
  margin-top: 4px;
  margin-left: 8px;
}

.expanderIcon{
  color: var(--mdc-theme-primary);
}
