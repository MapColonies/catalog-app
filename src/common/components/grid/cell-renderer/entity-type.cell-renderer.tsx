import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { IconButton, Tooltip } from '@map-colonies/react-core';
import { Box } from '@map-colonies/react-components';
import { ILayerImage } from '../../../../discrete-layer/models/layerImage';
import { LayerRasterRecordModelType } from '../../../../discrete-layer/models';

interface IEntityTypeCellRendererParams extends ICellRendererParams {
  style?: Record<string, unknown>;
}

export const EntityTypeRenderer: React.FC<IEntityTypeCellRendererParams> = (props) => {
  const type = (props.data as ILayerImage).__typename;
  let icon: string;
  let includedInBests;
  let value = '';
  switch (type) {
    case 'LayerRasterRecord':
      includedInBests = (props.data as LayerRasterRecordModelType).includedInBests;
      icon = includedInBests ? 'mc-icon-Map-Best-Orthophoto' : 'mc-icon-Map-Orthophoto';
      value = `Orthophoto ${includedInBests ? (' included in: ' + includedInBests.join(' , ')) : ''}`;
      break;
    case 'Layer3DRecord':
      icon = 'mc-icon-Map-3D';
      value = '3D';
      break;
    case 'BestRecord':
      icon = 'mc-icon-Bests';
      value = 'BEST';
      break;
    default:
      icon = '';
  }

  return (
    <Box style={props.style}>
      <Tooltip content={value}>
        <IconButton className={icon} label="ENTITY TYPE ICON"/>
      </Tooltip>
    </Box>
  );
};
