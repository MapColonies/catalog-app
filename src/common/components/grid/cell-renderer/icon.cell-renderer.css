.cellRendererIcon {
  color: var(--mdc-theme-primary) !important;
  position: relative;
  top: 4px;
  right: 16px;
}

body[dir='rtl'] .cellRendererIcon {
  right: unset;
  left: 16px;
}
