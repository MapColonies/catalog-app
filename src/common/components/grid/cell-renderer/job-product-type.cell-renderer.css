.productTypeIconContainer {
    display: flex;
    justify-content: flex-end;
}