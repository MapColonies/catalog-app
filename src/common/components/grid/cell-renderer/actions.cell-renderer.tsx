import React, { useState } from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { isEmpty } from 'lodash';
import { IconButton, MenuSurfaceAnchor, Typography, Menu, MenuItem } from '@map-colonies/react-core';
import { Box } from '@map-colonies/react-components';
import { ILayerImage } from '../../../../discrete-layer/models/layerImage';
import { IActionGroup, IAction } from '../../../actions/entity.actions';

import './actions.cell-renderer.css';

interface IActionsRendererParams extends ICellRendererParams {
  actions: IActionGroup[];
  actionHandler: (action: Record<string,unknown>) => void;
}

export const ActionsRenderer: React.FC<IActionsRendererParams> = (props) => {
  const entity = (props.data as ILayerImage).__typename as string;
  // @ts-ignore
  const actions = props.actions[entity] as IActionGroup[];
  let frequentActions: IAction[] = [];
  let allFlatActions: IAction[] = [];
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
  if (actions !== undefined) {
    actions.forEach(actionGroup => {
      frequentActions = [
        ...frequentActions,
        ...actionGroup.group.filter(action => action.frequent)
      ];
      allFlatActions = [
        ...allFlatActions,
        ...actionGroup.group
      ];
    });
  }

  const [openActionsMenu, setOpenActionsMenu] = useState<boolean>(false);

  const sendAction = (entity: string, action: IAction, data: Record<string,unknown>): void => {
    console.log(`SEND ${action.action} EVENT`);
    props.actionHandler({
      action: `${entity}.${action.action}`,
      data: data,
    });
  }

  return (
    <Box id="actionsCellRenderer" className="actionsContainer">
      {
        frequentActions.map((action, idx) => {
          return (
            <IconButton
              className={action.class ? `actionIcon actionDismissible ${action.class}` : `actionIcon actionDismissible`}
              icon={action.icon}
              key={`freqAct_${(props.data as ILayerImage).id}_${idx}`}
              onClick={(evt): void => { 
                sendAction(entity, action, props.data);
              }}
            />
          );
        })
      }
      <MenuSurfaceAnchor id="gridActionsMenuContainer">
        <Menu
          open={openActionsMenu}
          onClose={(evt): void => setOpenActionsMenu(false)}
          onMouseOver={(evt): void => evt.stopPropagation()}
        >
          {
            allFlatActions.map((action, idx) => {
              return (
                <>
                  <MenuItem key={`menuItemAct_${(props.data as ILayerImage).id}_${idx}`}>
                    <Box 
                      onClick={(evt): void => {
                        sendAction(entity, action, props.data);
                        setOpenActionsMenu(false); 
                      }}
                      className="actionMenuItem"
                    >
                      <IconButton
                        className={action.class ? `actionIcon actionDismissible ${action.class}` : `actionIcon actionDismissible`}
                        icon={action.icon}
                      />
                      <Typography 
                        tag="div"
                        className="actionMenuItemTitle actionDismissible"
                      >
                        {action.titleTranslationId}
                      </Typography>
                    </Box>
                  </MenuItem>
                  {idx === 3 && <hr/>}
                </>
              );
            })
          }
        </Menu>
        {
          !isEmpty(allFlatActions) &&
          <IconButton 
            id="allActionsIcon"
            icon="more_vert" 
            className="actionIcon" 
            onClick={(evt): void => setOpenActionsMenu(!openActionsMenu)}
          />
        }
      </MenuSurfaceAnchor>
    </Box>
  );
};
