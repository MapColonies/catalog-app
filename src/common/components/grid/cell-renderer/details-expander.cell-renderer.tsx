import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import {CollapseButton} from '../../collapse-button/collapse.button';
import { DETAILS_ROW_ID_SUFFIX, IGridRowDataDetailsExt } from '../grid';

import './details-expander.cell-renderer.css';

interface DetailsExpanderRendererProps extends ICellRendererParams {
  detailsRowCellRendererPresencePredicate?: (data: any) => boolean;
}

export const DetailsExpanderRenderer: React.FC<DetailsExpanderRendererProps> = (props): JSX.Element | null => {
  const shouldRenderBtn = props.detailsRowCellRendererPresencePredicate?.(props.data) ?? true;

  const handleCollapseExpand = (): void => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    const rowNode = props.api.getRowNode(`${props.data?.id as string}${DETAILS_ROW_ID_SUFFIX}`);   
    const isVisible = (rowNode.data as IGridRowDataDetailsExt).isVisible;
    rowNode.setDataValue('isVisible', !isVisible);
    props.api.onFilterChanged();
  };
  if (shouldRenderBtn) return  <CollapseButton onClick={handleCollapseExpand} />

  return null;
};
