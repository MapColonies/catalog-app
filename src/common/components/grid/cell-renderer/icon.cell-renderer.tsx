import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { Icon } from '@map-colonies/react-core';

import './icon.cell-renderer.css';

export const IconRenderer: React.FC<ICellRendererParams> = (props) => {

  return (
    <>
    {
      props.value && <Icon 
        className="cellRendererIcon"
        icon={{ icon: 'star_rate', size: 'small' }}
      />
    }
    </>
  );

};
