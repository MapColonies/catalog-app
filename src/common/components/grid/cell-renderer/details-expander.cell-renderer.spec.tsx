import React from 'react';
import { shallow } from 'enzyme';
import { ICellRendererParams, Column, RowNode, GridApi, ColumnApi } from 'ag-grid-community';
// eslint-disable-next-line
import '../../../../__mocks__/confEnvShim';
import { DETAILS_ROW_ID_SUFFIX } from '../grid';
import { DetailsExpanderRenderer } from './details-expander.cell-renderer';

const ID = '1';
/* eslint-disable */
const mockDataBase:ICellRendererParams = {
  value: '',
  valueFormatted: null,
  getValue: () => {},
  setValue: () => {},
  formatValue: () => {},
  data: {
    isVisible:true,
    id: ID, 
  },
  node: new RowNode(),
  colDef: {},
  column: new Column({},null,'isVisible',true),
  $scope: null,
  rowIndex: 1,
  api: new GridApi(),
  columnApi: new ColumnApi(),
  context: null,
  refreshCell: () => {},
  eGridCell: document.createElement('span'),
  eParentOfValue: document.createElement('span'),
  addRenderedRowListener: () => {},
};
/* eslint-enable */

describe('AgGrid DetailsExpanderRenderer component', () => {
  it('renders correctly', () => {
    const mockData = {
      ...mockDataBase
    };

    const wrapper = shallow(
      <DetailsExpanderRenderer {...mockData} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('when component clicked detail row was found in grid rowdata', () => {
    const mockData = {
      ...mockDataBase,
    };

    // eslint-disable-next-line
    jest.spyOn(mockData.api, 'onFilterChanged').mockImplementation(() => {});
    const spyGetRonNode = jest.spyOn(mockData.api, 'getRowNode').mockImplementation(() => {
      const val = new RowNode();
      // eslint-disable-next-line
      val.setDataValue = (propName, val) => {};
      val.data = {
        isVisible: false
      };
      return val;
    });

    const wrapper = shallow(
      <DetailsExpanderRenderer {...mockData} />
    );
    
    const iconContainer = wrapper.find('CollapseButton');
    iconContainer.simulate('click');
    expect(spyGetRonNode).toHaveBeenCalledWith(`${ID}${DETAILS_ROW_ID_SUFFIX}`);
  });

});
