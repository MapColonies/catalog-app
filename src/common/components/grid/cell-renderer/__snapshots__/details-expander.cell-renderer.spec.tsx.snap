// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgGrid DetailsExpanderRenderer component renders correctly 1`] = `
<CollapseButton
  onClick={[Function]}
/>
`;
